examples/gno.land/p/gov/executor/callback.gno

```diff
@@ -0,0 +1,51 @@
+package executor
+
+import (
+	"errors"
+	"std"
+)
+
+var errInvalidCaller = errors.New("invalid executor caller")
+
+// NewCallbackExecutor creates a new callback executor with the provided callback function
+func NewCallbackExecutor(callback func() error, path string) *CallbackExecutor {
+	return &CallbackExecutor{
+		callback:   callback,
+		daoPkgPath: path,
+	}
+}
+
+// CallbackExecutor is an implementation of the dao.Executor interface,
+// based on a specific callback.
+// The given callback should verify the validity of the govdao call
+type CallbackExecutor struct {
+	callback   func() error // the callback to be executed
+	daoPkgPath string       // the active pkg path of the govdao
+}
+
+// Execute runs the executor's callback function.
+func (exec *CallbackExecutor) Execute() error {
+	// Verify the executor is r/gov/dao
+	assertCaller(exec.daoPkgPath)
```

Comment:
here you panic, after you return an error, maybe you should always return an error to be consistent.

Reply:
I agree, it's cleaner to return, since `Execute` already supports an error:

[676b202](https://github.com/gnolang/gno/pull/2581/commits/676b2021578446fc945de1b506aa2793e5595078)